nearswap/src/twap.rs

```diff
@@ -0,0 +1,553 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::json_types::{U128, U64};
+use near_sdk::collections::{Vector};
+use near_sdk::{env};
+use std::convert::{TryFrom,TryInto};
+
+use std::fmt;
+use crate::constants::*;
+use crate::pool::*;
+use crate::*;
+
+#[derive(Debug)]
+pub enum Mean {
+    M_1MIN,
+    M_5MIN,
+    M_1H,
+    M_12H,
+}
+
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct Twap {
+    // To check if array is populated or not
+    // note: it can be used later when we will have variable length storage array/
+    populated: usize,
+    // last updated index in observation array
+    last_updated_index: usize,
+    // bool to check if previous values are overwritten by new one
+    // i.e MAX_LENGTH of array is full and we start storing observation from `0` index
+    pivoted: bool,
+    // observation array
+    pub observations: Vec<Observation>,
+
+    // mean of last 1 minutes of readings
+    pub mean_1min: (U128, U128),
```

Comment:
please explain what are the tuple values.

Reply:
ok